pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -322,6 +331,25 @@ func (b *OFBridge) MultipartReply(sw *ofctrl.OFSwitch, rep *openflow15.Multipart
 	if ch, ok := b.multipartReplyChs[rep.Xid]; ok {
 		ch <- rep
 	}
+	if rep.Type == openflow15.MultipartType_MeterStats {
+		statsArr := rep.Body
+		for _, entry := range statsArr {
+			stats := entry.(*openflow15.MeterStats)
+			idLabel := ""
+			if stats.MeterId == 1 {
```

Comment:
these constants (1 & 2) are defined somewhere as `PacketInMeterIDNP` and `PacketInMeterIDTF` IIRC, so we should probably use them

Reply:
Yes, these two are originally defined in https://github.com/antrea-io/antrea/blob/main/pkg/agent/openflow/packetin.go.
But if I import them here, there will be an import cycle (pkg/ovs/openflow -> pkg/agent/openflow -> pkg/ovs/openflow). So I use 1 and 2 here to void the import cycle.

Reply:
Even if we ignore the matter of the constants, I think there is room for this code to be improved:
- leverage `multipartReplyChs` and `rep.Xid` like we do for table features
- have a more generic approach, with `GetMeterStats` taking a callback function

Something like:
```go
func (b *OFBridge) GetMeterStats(cb func(packetCounts map[int]int64, byteCounts map[int]int64)) error {
	const OFPM_ALL = 0xffffffff // Represents all meters
	mpMeterStatsReq :=
		openflow15.NewMpRequest(openflow15.MultipartType_MeterStats)
	meterMPReq := openflow15.NewMeterMultipartRequest(OFPM_ALL)
	mpMeterStatsReq.Body = append(mpMeterStatsReq.Body, meterMPReq)

        // do we need a larger buffer?
        ch := make(chan *openflow15.MultipartReply, 1)
        b.multipartReplyChs[mpMeterStatsReq.Xid] = ch
        go func() {
                defer func() {
                        delete(b.multipartReplyChs, mpMeterStatsReq.Xid)
                }()
                packetCounts := make(map[int]int64)
                byteCounts := make(map[int]int64)
                // process reply from channel here and update the 2 maps
                // ...
                cb(packetCounts, byteCounts)
        }

	return b.ofSwitch.Send(mpMeterStatsReq)
}
```

I am adding @wenyingd for her feedback on this.
It seems a bit strange to me that we put so much special logic into the `MultipartReply` function. Ideally this function should not need to know about which meters have been defined in the OVS pipeline.

Reply:
I agree with @antoninbas  to not place logic for parsing the meter details in `MultipartReply`. MultipartRequest/Reply is a general mechanism in OF messages, which is possible to use for different scenarios. It will introduce difficulty for future maintanance if we place parser details here.

Reply:
I don't like the idea to define function related constants in path "pkg/ovs", it means we need to update here once the usage of Meter is changed. Maybe we could think about add a new utility to map the OVS resource and its function meaning?